resource.c

```diff
@@ -231,17 +239,23 @@ int res_read_blk(res_blk_t *res, int pid, int flags)
 			break;
 
 		case RES_MEM_PAGESIZE:
-			(res->res_unit[i]->data).sz = sysconf(_SC_PAGESIZE);
-			res->res_unit[i]->status = RES_STATUS_FILLED;
+			if (res->res_unit[i]->data_sz < sizeof(long)) {
+				res->res_unit[i]->status = ENOMEM;
+			} else {
+				(res->res_unit[i]->data).sz = sysconf(_SC_PAGESIZE);
+				res->res_unit[i]->status = RES_STATUS_FILLED;
+			}
 			break;
 
 		case RES_KERN_RELEASE:
 			ret = uname(&t);
 			if (ret == -1) {
 				res->res_unit[i]->status = errno;
 			} else {
-				strncpy((res->res_unit[i]->data).str,
-					t.release, RESOURCE_64);
+				out = (res->res_unit[i]->data).str;
+				len = sizeof(union r_data);
+				strncpy(out, t.release, len-1);
```

Comment:
In this case this probably can't happen, but as a code pattern goes, the problem here is that if len > length of the string in t.release, then you may leak uninitialized memory here.